library/kani_macros/src/sysroot/contracts/mod.rs

```diff
@@ -67,61 +64,39 @@
 //!                     │ Function  │
 //!                     └─────┬─────┘
 //!                           │
-//!            Emit           │  Generate      + Copy Attributes
-//!         ┌─────────────────┴─────┬──────────┬─────────────────┐
-//!         │                       │          │                 │
-//!         │                       │          │                 │
-//!         ▼                       ▼          ▼                 ▼
-//!  ┌──────────┐           ┌───────────┐  ┌───────┐        ┌─────────┐
-//!  │ Original │◄─┐        │ Recursion │  │ Check │◄─┐     │ Replace │◄─┐
-//!  └──┬───────┘  │        │ Wrapper   │  └───┬───┘  │     └────┬────┘  │
-//!     │          │ Ignore └───────────┘      │      │ Augment  │       │ Augment
-//!     └──────────┘                           └──────┘          └───────┘
-//!
-//! │               │       │                                             │
-//! └───────────────┘       └─────────────────────────────────────────────┘
-//!
-//!     Presence of                            Presence of
-//!    "checked_with"                    "is_contract_generated"
-//!
-//!                        State is detected via
+//!                           │  Annotate original + generate closures
+//!                           │
+//!                           ▼
+//!                    ┌──────────┐
+//!                    │ Original │◄─┐
+//!                    └──┬───────┘  │
+//!                       │          │ Expand
+//!                       └──────────┘ closures
```

Comment:
What happened here? I think this new diagram version is not very helpful. We still generate different types of functions (closures in this case) and use marker annotations to distinguish how a particular clause modifies code so I think that should be reflected here. Especially now since this code generation is getting increasingly complex.

Reply:
I was actually tempted to remove this, since there is no state tracking. We have 2 states now, untouched and touched. In that sense, the state machine got simpler.

I agree that modifies and old complicates the code generation a bit though, but maybe we can keep the documentation more local so it's easier to maintain.